include/nudb/impl/basic_store.ipp

```diff
@@ -233,20 +232,25 @@ insert(
     std::lock_guard<std::mutex> u{u_};
     {
         shared_lock_type m{m_};
-        if(s_->p1.find(key) != s_->p1.end())
-            return false;
-        if(s_->p0.find(key) != s_->p0.end())
-            return false;
+        if(s_->p1.find(key) != s_->p1.end() ||
+           s_->p0.find(key) != s_->p0.end())
+        {
+            ec = error::key_exists;
+            return;
+        }
         auto const n = bucket_index(h, buckets_, modulus_);
         auto const iter = s_->c1.find(n);
         if(iter != s_->c1.end())
         {
             auto const found = exists(
                 h, key, &m, iter->second, ec);
             if(ec)
```

Comment:
If exists set ec if the key doesn't exist, this could be
```
exists(h, key, &m, iter->second, ec);
if(! ec)
{
	ec = error::key_exists;
	return;
}
```